core/server/common/src/main/java/alluxio/master/PrimarySelectorClient.java

```diff
@@ -90,15 +93,11 @@ public PrimarySelectorClient(String zookeeperAddress, String electionPath, Strin
   }
 
   @Override
-  public void close() throws IOException {
-    try {
+  public synchronized void close() throws IOException {
+    if (mLifecycleState == LifecycleState.STARTED) {
```

Comment:
Could you move this logic to `stop()` after making sure `close()` is not called directly?
So that state transitions make more sense to readers.

Reply:
I removed the `close()` method.